pkg/utils/maputils.go

```diff
@@ -84,3 +86,37 @@ func SetDifference[K comparable, T1, T2 any](newSet map[K]T2, oldSet map[K]T1) (
 
 	return added, removed
 }
+
+// ConvertToSetFromMap converts a map to a set. The set is represented as a map with the key as the element of the set
+func ConvertToSetFromMap[K comparable, V any](set map[K]struct{}, source map[K]V) {
+	for k := range source {
+		set[k] = struct{}{}
+	}
+}
+
+// ConvertToSetFromSlice converts a slice to a set. The set is represented as a map with the key as the element of the set
+func ConvertToSetFromSlice[K comparable](set map[K]struct{}, source []K) {
+	for _, k := range source {
+		set[k] = struct{}{}
+	}
+}
```

Comment:
nit:
1. `ConvertMapToSet` and `ConvertSliceToSet` would be slightly more concise, but you can keep the `ConvertToXFromY` format if you want
2. Using `Add` instead of `Convert` in the function name seems more precise
3. For `ConvertToSetFromMap()` it would be good to specify in the name that you're adding the map keys rather than the values